gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql

```diff
@@ -0,0 +1,109 @@
+/* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql */
+
+/*
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
+RETURNS VOID
+AS $$
+BEGIN
+	IF NOT EXISTS (
+		SELECT
+		FROM pg_partition pp
+		JOIN pg_partition_rule pr ON pr.paroid = pp.oid
+		WHERE
+			CASE
+				WHEN pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
+				THEN now() BETWEEN

```

Comment:
now() returns `TIMESTAMP WITH TIME ZONE`. Is it correct to compare it with `TIMESTAMP WITHOUT TIME ZONE`?

Reply:
From https://www.postgresql.org/docs/current/functions-datetime.html
> Dates and timestamps (with or without time zone) are all comparable, while times (with or without time zone) and intervals can only be compared to other values of the same data type. When comparing a timestamp without time zone to a timestamp with time zone, the former value is assumed to be given in the time zone specified by the [TimeZone](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-TIMEZONE) configuration parameter, and is rotated to UTC for comparison to the latter value (which is already in UTC internally)

Thus I think yes, it is Ok to compare.

Reply:
After some thinking, I guess the only problem that may happen if the TimeZone parameter changes at some point, in this case the next split of partition might occur a bit earlier or later. But that should not cause any data loss or any other critical thing, and it shouldn't impact the results of select operation applied to this table. And changing this will require changing the `db_files_history` table structure, which I think we should try to avoid in this PR.

Reply:
may be use
```sql
now()::timestamp
```
?

Reply:
> may be use

What will it give? 

And let's first define what problem we are trying to resolve here. Right now I see the only issue that could occur if the timezone is changed, and it actually should not affect much the user from my point of view.